fast64_internal/sm64/sm64_objects.py

```diff
@@ -1403,6 +1421,719 @@ def draw_act(self, obj, layout, value):
         layout.prop(obj, "sm64_obj_use_act" + str(value), text="")
 
 
+class AddBehavior(bpy.types.Operator):
+    bl_idname = "bone.add_behavior_script"
+    bl_label = "Add Behavior Script"
+    option: bpy.props.IntProperty()
+
+    def execute(self, context):
+        prop = context.scene.fast64.sm64.combined_export
+        prop.behavior_script.add()
+        prop.behavior_script.move(len(prop.behavior_script) - 1, self.option)
+        self.report({"INFO"}, "Success!")
+        return {"FINISHED"}
+
+
+class RemoveBehavior(bpy.types.Operator):
+    bl_idname = "bone.remove_behavior_script"
+    bl_label = "Remove Behavior Script"
+    option: bpy.props.IntProperty()
+
+    def execute(self, context):
+        prop = context.scene.fast64.sm64.combined_export
+        prop.behavior_script.remove(self.option)
+        self.report({"INFO"}, "Success!")
+        return {"FINISHED"}
+
+
+class BehaviorScriptProperty(bpy.types.PropertyGroup):
+    expand: bpy.props.BoolProperty(name="Expand")
+    macro: bpy.props.EnumProperty(items=enumBehaviorMacros, name="Behavior Macro", default="BEGIN")
+    # there are anywhere from 0 to 3 arguments for a bhv, rather than make a new prop
+    # group and collection prop, I'll just have static props and choose to use
+    # arguments as needed
+    arg_1: bpy.props.StringProperty(name="Argument 1")
+    arg_2: bpy.props.StringProperty(name="Argument 2")
+    arg_3: bpy.props.StringProperty(name="Argument 3")
+    # for gravity, 8 args are used, but everything else only uses a max of 3
+    arg_4: bpy.props.StringProperty(name="Argument 4")
+    arg_5: bpy.props.StringProperty(name="Argument 5")
+    arg_6: bpy.props.StringProperty(name="Argument 6")
+    arg_7: bpy.props.StringProperty(name="Argument 7")
+    arg_8: bpy.props.StringProperty(name="Argument 8")
+
+    # some objects have cmds that make sense to dynamically inherit it from export properties
+    # load collision,  or set model ID are easy examples
+    inherit_from_export: bpy.props.BoolProperty(name="Inherit From Export")
+
+    _inheritable_macros = {
+        "LOAD_COLLISION_DATA",
+        "SET_MODEL",
+        # add support later maybe
+        # "SET_HITBOX_WITH_OFFSET",
+        # "SET_HITBOX",
+        # "SET_HURTBOX",
```

Comment:
How would this work? Do you plan on adding an automatic hitbox and hurtbox or something? I do think we should have something like that for culling radius too, I have logic for an auto culling radius in an old branch of mine and it works great!